### ISSUE TITLE
Error not thrown if given a bad site

### ISSUE THREAD

**Issue by bradvogel** [shinnn/image-size-stream#3]
In my app I can't always predict if a URL is a valid image. For some urls such as http://mobilefuture.org/cisco-projects-the-future-of-mobile/ the stream service doesn't emit a 'size' or 'error' event at all. I would expect it to always emit one or the other.

```
var http = require('http');

var createImageSizeStream = require('image-size-stream');
var size = createImageSizeStream();
size
  .on('size', function(dimensions) {
    console.log('response', dimensions)
    request.abort();
  })
  .on('error', function(err) {
    console.log('error', err);
    throw err;
  });

var request = http.get('http://mobilefuture.org/cisco-projects-the-future-of-mobile/', function(response) {
  response.pipe(size);
});
```

**Comment by shinnn** [comment]
Thanks for reporting. Currently I cannot show you an essential solution for this problem.

For now, maybe [http-image-size-promise](https://github.com/shinnn/http-image-size-promise) module helps you.

**Comment by bradvogel** [comment]
Ok, what's the root cause though? Is it something fundamentally broken?

**Comment by shinnn** [comment]
I found the cause. This code works as you expected.

```js
var http = require('http');

var createImageSizeStream = require('image-size-stream');
var size = createImageSizeStream();
size
  .on('size', function(dimensions) {
    console.log('response', dimensions)
    request.abort();
  })
  .on('error', function(err) {
    console.log('error', err);
    throw err;
  });

var request = http.get('http://mobilefuture.org/cisco-projects-the-future-of-mobile/', function(response) {
  response.pipe(size);
  response.on('end', function() {
    size.emit('finish');
  });
});
```

Even if the response stream ends, image-size-stream doesn't finish until it reaches the [limit](https://github.com/shinnn/image-size-stream#optionlimit). So we should emit `finish` event explicitly when response stream ends.

I'll update readme.

**Comment by bradvogel** [comment]
Oh, thanks! Yes, a readme update would be great.

**Comment by shinnn** [comment]
I released v1.1.0 that doesn't require [this workaround](https://github.com/shinnn/image-size-stream/issues/3#issuecomment-75920826).
